src/main/java/org/broadinstitute/consent/http/service/DatasetService.java

```diff
@@ -388,7 +388,13 @@ public void deleteDataset(Integer datasetId, Integer userId) throws Exception {
                     ds.getConsentId().toLowerCase().contains(lowercasePartial) ||
                     ds.getProperties().stream()
                           .anyMatch(
-                                p -> p.getPropertyValue().toLowerCase().contains(lowercasePartial))
+                                p -> {
+                                    String value = p.getPropertyValue();
+                                    if(Objects.isNull(value)) {
+                                        value = "";
+                                    }
+                                    return value.toLowerCase().contains(lowercasePartial);
+                                })
```

Comment:
A simpler way to do this is just to filter out the nulls before trying to examine them, that way you know you won't hit the error. In this solution, the empty string should match any value provided which looks like it might be giving you too many positive results.

This should work:
```
                    ds.getProperties().stream()
                            .filter(Objects::nonNull)
                            .filter(p -> Objects.nonNull(p.getPropertyValue()))
                            .anyMatch(p -> p.getPropertyValue().toLowerCase().contains(lowercasePartial))
```
Additionally, just above this, we have the same problem on line 386. I think it might be better to do the filtering before either of these checks happen, probably right after line 383.